relays/lib-substrate-relay/src/lib.rs

```diff
@@ -90,50 +91,41 @@ impl<AccountId> TaggedAccount<AccountId> {
 }
 
 /// Batch call builder.
-pub trait BatchCallBuilder<Call> {
-	/// Associated error type.
-	type Error;
-	/// If `true`, then batch calls are supported at the chain.
-	const BATCH_CALL_SUPPORTED: bool;
-
+pub trait BatchCallBuilder<Call>: Send {
 	/// Create batch call from given calls vector.
-	fn build_batch_call(_calls: Vec<Call>) -> Result<Call, Self::Error>;
+	fn build_batch_call(&self, _calls: Vec<Call>) -> Call;
 }
 
-impl<Call> BatchCallBuilder<Call> for () {
-	type Error = SubstrateError;
-	const BATCH_CALL_SUPPORTED: bool = false;
+/// Batch call builder constructor.
+pub trait BatchCallBuilderConstructor<Call> {
```

Comment:
I won't insist on changing that, but I'd like to note that imo 4 traits (`BatchCallBuilderConstructor`, `BatchCallBuilder`, `ChainWithUtilityPallet`, `UtilityPallet`) instead of single one (`BatchCallBuilder`) is too much :) E.g. which other implementations of the `UtilityPallet` we may have? :) But as I said - if it solves the problem, let's keep it as is :)

Reply:
Yes, I don't like to have so many traits either, but I couldn't think of another solution. I think that `ChainWithUtilityPallet` makes sense since this way we can associate the batching functionality with the chain. And also we can add constraints like implementing `UtilityPalletBatchCallBuilder` only for chains with utility pallet. And adding `BatchCallBuilderConstructor` is a small compromise for safety. About the `UtilityPallet` trait I'm not sure, but from what I understand we will have 2 implementations for it. One for the chains that are defined in `parity-bridges-common` and for which we have the full runtime (Rialto, Millau). And one for the chains defined outside (the RococoBridgeHub, Wococo bridgeHub for example) where we only have a RuntimeWrapper. I might be wrong. It would be nice if we can avoid it.